src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -127,6 +132,9 @@ public async Task<BundleResponse> Handle(BundleRequest bundleRequest, Cancellati
             var bundleResource = bundleRequest.Bundle.ToPoco<Hl7.Fhir.Model.Bundle>();
             _bundleType = bundleResource.Type;
 
+            // For resources within a transaction, we need to validate if they are referring to each other and throw an exception in such case.
+            await _transactionValidator.ValidateBundle(bundleResource);
```

Comment:
The comment says within a transaction, but it will run for both batch and transaction, which is valid?

Reply:
it runs for transaction only. The method ValidateBundle checks resourceType before proceeding.